packages/compiler/src/lib/decorators.ts

```diff
@@ -405,21 +405,53 @@ export function getFormat(program: Program, target: Type): string | undefined {
 
 const patternValuesKey = createStateSymbol("patternValues");
 
+export interface PatternData {
+  pattern: string;
+  validationMessage?: string;
+}
```

Comment:
```suggestion
export interface PatternData {
  readonly pattern: string;
  readonly validationMessage?: string;
}
```
thinking we should start making things readonly in general as people will write more and more decorators,emitters and stuff changing that data will result in very unexpected result

Reply:
Good point. I'll integrate this suggestion. Do you have any thoughts about making AST node properties read-only after we finish them? We want to strongly discourage writing to the AST for similar reasons.